schwab_converter.py

```diff
@@ -45,20 +53,22 @@ def _check_header(self):
                             "FairMarketValuePrice", "SalePrice", "SharesSoldWithheldForTaxes", "NetSharesDeposited",
                             "Taxes", "GrossProceeds"]
 
-        if self.row != expected_headers:
-            raise ValueError("Schwab CSV is not in the expected format. Either this script needs adaption or "
-                             "a wrong type of CSV was downloaded.")
-
+        if set(self.row) < set(expected_headers):
+            raise ValueError(f"Schwab CSV is missing some columns: {set(expected_headers) - set(self.row)}")
+        
+        # CSV export might map to different headers
+        self.mapping = _create_mapping(expected_headers, self.row)
+        
         self.header_checked = True
 
     def _process_trade_row(self):
-        if self.row[1] in ["Deposit", "Sale"]:
-            date = datetime.strptime(self.row[0], "%m/%d/%Y").date()  # Date
-            symbol = self.row[2]  # Symbol
-            commission = self._parse_usd('0' if not self.row[5] else self.row[5])  # FeesAndCommissions
-            quantity = self._parse_number(self.row[4])  # Quantity
+        if self.row[self.mapping[1]] in ["Deposit", "Sale"]:
```

Comment:
Please, create a new function something like below to have all this repeated boilerplate in one place, so it'll be easier to read and modify later:

```python
def _field(self, index: int):
    return self.row[self.mapping[index]]
```

Reply:
Since this function is in place, the variable name `mapping` is not anymore length-bound, could you please use `column_mapping` instead to make it more obvious what kind of mapping we're talking about?